addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -1032,6 +1028,23 @@ odoo.define('pos_coupon.pos', function (require) {
             });
             return [discountRewards, discountRewards.length > 0 ? null : 'No items to discount.'];
         },
+
+        _createFixedDiscountRewards: function (program, coupon_id, amountsToDiscount) {
+            const discountAmount = Math.min(program.discount_fixed_amount, program.discount_max_amount || Infinity);
+            const discountRatio = discountAmount / this.get_total_with_tax();
+            const discountRewards = Object.entries(amountsToDiscount).map(([tax_keys, amount]) => {
+                let discountAmount = amount * discountRatio;
```

Comment:
Try a product with unit price $100 with two taxes -- 15% and 20%. Then apply a fixed discount of $20. The resulting discount that will be applied is $19.99 (with tax). I'm thinking it's because of the `get_total_with_tax` which, if I'm not mistaken, is rounded. (To see, make sure to activate tax-included in the pos.config.)

Can you try to find a way to resolve this? Maybe there is a way.